parsl/executors/errors.py

```diff
@@ -44,6 +46,17 @@ def __str__(self):
                                                                  self.current_executor)
 
 
+class InvalidResourceSpecificationError(ExecutorError):
```

Comment:
you could keep this name the same as before this PR, InvalidResourceSpecification without Error on the end. Many exception classes in Parsl omit `Error` or `Exception` on the end